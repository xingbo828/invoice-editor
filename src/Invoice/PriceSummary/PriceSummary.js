import React from 'react';

import {
  PriceSummaryOuterContainer,
  PriceSummaryInnerContainer,
  PriceSummaryLineItem
} from './styles';

const PriceSummary = ({ formatter, total, tax, subtotal }) => (
  <PriceSummaryOuterContainer>
    <PriceSummaryInnerContainer>
      <PriceSummaryLineItem>
        <span>Subtotal</span>
        <span>{formatter.format(subtotal)}</span>
      </PriceSummaryLineItem>
      <PriceSummaryLineItem>
        <span>Tax (5%)</span>
        <span>{formatter.format(tax)}</span>
      </PriceSummaryLineItem>
      <PriceSummaryLineItem>
        <span>Total</span>
        <span>{formatter.format(total)}</span>
      </PriceSummaryLineItem>
    </PriceSummaryInnerContainer>
  </PriceSummaryOuterContainer>
);

export default PriceSummary;
