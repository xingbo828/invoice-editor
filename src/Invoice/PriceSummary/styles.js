import styled from 'styled-components'

export const PriceSummaryOuterContainer = styled.div`
  font-size: 1.25rem;
  display: flex;
  flex-direction: row-reverse;
  max-width: 800px;
  margin: 0 auto;
`

export const PriceSummaryInnerContainer = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  ${props=>props.theme.media.greaterThan('sm')`
    width: 300px;
  `}
`

export const PriceSummaryLineItem = styled.div`
  display: flex;
  padding: .5rem 0;
  justify-content: space-between;
`