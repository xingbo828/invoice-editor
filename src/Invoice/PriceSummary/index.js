import { compose, withProps } from 'recompose'
import { connect } from 'react-redux'

import { getProductsTotal } from '../../cartReducer'


import PriceSummary from './PriceSummary'

const tax = 0.05

const formatter = new Intl.NumberFormat('en-US', {
  style: 'currency',
  currency: 'USD',
  minimumFractionDigits: 2
})

const mapStateToProps = (state) => ({
  subtotal: getProductsTotal(state)
});

const enhance = compose(
  connect(mapStateToProps),
  withProps(({ subtotal }) => ({
    tax: tax * subtotal,
    total: subtotal * (1 + tax),
    formatter
  }))
)


export default enhance(PriceSummary)