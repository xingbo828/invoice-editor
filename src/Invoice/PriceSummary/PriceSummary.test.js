import React from 'react';
import { shallow } from 'enzyme';
import PriceSummary from './PriceSummary';

const props = {
  subtotal: 100,
  tax: 5,
  total: 105,
  formatter: {
    format: p => p
  }
}
it('should match snapshot', () => {
  const wrapper = shallow(<PriceSummary {...props} />);
  expect(wrapper).toMatchSnapshot()
});