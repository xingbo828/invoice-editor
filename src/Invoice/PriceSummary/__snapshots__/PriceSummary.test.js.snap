// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PriceSummary
    formatter={
      Object {
        "format": [Function],
      }
    }
    subtotal={100}
    tax={5}
    total={105}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ForwardRef>
        <ForwardRef>
          <span>
            Subtotal
          </span>
          <span>
            100
          </span>
        </ForwardRef>
        <ForwardRef>
          <span>
            Tax (5%)
          </span>
          <span>
            5
          </span>
        </ForwardRef>
        <ForwardRef>
          <span>
            Total
          </span>
          <span>
            105
          </span>
        </ForwardRef>
      </ForwardRef>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef>
            <span>
              Subtotal
            </span>
            <span>
              100
            </span>
          </ForwardRef>,
          <ForwardRef>
            <span>
              Tax (5%)
            </span>
            <span>
              5
            </span>
          </ForwardRef>,
          <ForwardRef>
            <span>
              Total
            </span>
            <span>
              105
            </span>
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <span>
                Subtotal
              </span>,
              <span>
                100
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Subtotal",
              },
              "ref": null,
              "rendered": "Subtotal",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 100,
              },
              "ref": null,
              "rendered": 100,
              "type": "span",
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-htpNat",
              "isStatic": true,
              "rules": Array [
                "
  display: flex;
  padding: .5rem 0;
  justify-content: space-between;
",
              ],
            },
            "displayName": "styled.div",
            "render": [Function],
            "styledComponentId": "sc-htpNat",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <span>
                Tax (5%)
              </span>,
              <span>
                5
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Tax (5%)",
              },
              "ref": null,
              "rendered": "Tax (5%)",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 5,
              },
              "ref": null,
              "rendered": 5,
              "type": "span",
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-htpNat",
              "isStatic": true,
              "rules": Array [
                "
  display: flex;
  padding: .5rem 0;
  justify-content: space-between;
",
              ],
            },
            "displayName": "styled.div",
            "render": [Function],
            "styledComponentId": "sc-htpNat",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <span>
                Total
              </span>,
              <span>
                105
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Total",
              },
              "ref": null,
              "rendered": "Total",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 105,
              },
              "ref": null,
              "rendered": 105,
              "type": "span",
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-htpNat",
              "isStatic": true,
              "rules": Array [
                "
  display: flex;
  padding: .5rem 0;
  justify-content: space-between;
",
              ],
            },
            "displayName": "styled.div",
            "render": [Function],
            "styledComponentId": "sc-htpNat",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": false,
          "rules": Array [
            "
  width: 100%;
  display: flex;
  flex-direction: column;
  ",
            [Function],
            "
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  font-size: 1.25rem;
  display: flex;
  flex-direction: row-reverse;
  max-width: 800px;
  margin: 0 auto;
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef>
          <ForwardRef>
            <span>
              Subtotal
            </span>
            <span>
              100
            </span>
          </ForwardRef>
          <ForwardRef>
            <span>
              Tax (5%)
            </span>
            <span>
              5
            </span>
          </ForwardRef>
          <ForwardRef>
            <span>
              Total
            </span>
            <span>
              105
            </span>
          </ForwardRef>
        </ForwardRef>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef>
              <span>
                Subtotal
              </span>
              <span>
                100
              </span>
            </ForwardRef>,
            <ForwardRef>
              <span>
                Tax (5%)
              </span>
              <span>
                5
              </span>
            </ForwardRef>,
            <ForwardRef>
              <span>
                Total
              </span>
              <span>
                105
              </span>
            </ForwardRef>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <span>
                  Subtotal
                </span>,
                <span>
                  100
                </span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Subtotal",
                },
                "ref": null,
                "rendered": "Subtotal",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 100,
                },
                "ref": null,
                "rendered": 100,
                "type": "span",
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-htpNat",
                "isStatic": true,
                "rules": Array [
                  "
  display: flex;
  padding: .5rem 0;
  justify-content: space-between;
",
                ],
              },
              "displayName": "styled.div",
              "render": [Function],
              "styledComponentId": "sc-htpNat",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <span>
                  Tax (5%)
                </span>,
                <span>
                  5
                </span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Tax (5%)",
                },
                "ref": null,
                "rendered": "Tax (5%)",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 5,
                },
                "ref": null,
                "rendered": 5,
                "type": "span",
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-htpNat",
                "isStatic": true,
                "rules": Array [
                  "
  display: flex;
  padding: .5rem 0;
  justify-content: space-between;
",
                ],
              },
              "displayName": "styled.div",
              "render": [Function],
              "styledComponentId": "sc-htpNat",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <span>
                  Total
                </span>,
                <span>
                  105
                </span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Total",
                },
                "ref": null,
                "rendered": "Total",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 105,
                },
                "ref": null,
                "rendered": 105,
                "type": "span",
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-htpNat",
                "isStatic": true,
                "rules": Array [
                  "
  display: flex;
  padding: .5rem 0;
  justify-content: space-between;
",
                ],
              },
              "displayName": "styled.div",
              "render": [Function],
              "styledComponentId": "sc-htpNat",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "rules": Array [
              "
  width: 100%;
  display: flex;
  flex-direction: column;
  ",
              [Function],
              "
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  font-size: 1.25rem;
  display: flex;
  flex-direction: row-reverse;
  max-width: 800px;
  margin: 0 auto;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
